File: cpp/questions/plusOne.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for (int p=digits.size()-1; p>=0; p--) {
            auto& d = digits[p];
            if (++d == 10) {
                d = 0;
            } else {
                return digits;
            }
        }
        digits.push_back(0);
        digits[0] = 1;
        return digits;
    }
};

File: python/questions/plusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        for p in range(len(digits)-1, -1, -1):
            if digits[p] < 9:
                digits[p] += 1
                return digits
            else:
                digits[p] = 0
        digits.append(0)
        digits[0] = 1
        return digits
